addons/web/static/src/views/fields/badge/badge_field.scss

```diff
@@ -0,0 +1,3 @@
+.o_field_badge span {
+    background-color: $o-gray-300;
```

Comment:
I have made the changes. what do you think, is this a good way to handle the background color?

Thank you for your suggestion.

Reply:
I would apply a `text-bg-300` instead of `bg-300` sorry, so in case we change the gray-300 value in the future, the text will stay contrasted 

At first glance it seems correct but to be checked with @aab-odoo if we can return it directly there as a default value for badges that don't have a decoration props 😄 

Also there seems to be some hardcoded `o_field_badge` that were probably benefiting from the .scss rule, so we should check if they require the `text-bg-300` on them as well.
![image](https://github.com/user-attachments/assets/7e6db4b7-c6d4-4650-a377-3af439a1e900)

Thanks again!

Reply:
@mano-odoo That makes sense indeed.

Reply:
I believe we should proceed with solution of external scss file, there are two usecase where we not getting text-bg-300 applied as suggested earlier
-  [mrp_mo_overview_line.xml](https://github.com/odoo/odoo/blob/288d3926c5d011590aa25e9886cfa36377974dd6/addons/mrp/static/src/components/mo_overview_line/mrp_mo_overview_line.xml#L17) 
- [product_pricelist_report.xml](https://github.com/odoo/odoo/blob/288d3926c5d011590aa25e9886cfa36377974dd6/addons/product/static/src/js/pricelist_report/product_pricelist_report.xml#L62) 
For the product pricelist report, we can directly apply the class to the badge. However, I couldn't find a better solution for the `mrp_mo_overview_line` case.
Additionally, there are some issues with the current solution where text-bg-muted is not being applied if it is already defined. For example, as shown in the screenshot, the text-bg-muted decoration is given to the "Draft" state, but it doesn't get applied.
![image](https://github.com/user-attachments/assets/b1339e74-0b21-4705-b2f1-63f47562c859)


What do you think? If you have any suggestions, please let me know.

Reply:
Hello @aab-odoo 

The issue in both cases is due to the `text-bg-muted` class. It seems that text-bg-muted is no longer supported in our current version of Bootstrap.in version 17.0, text-bg-muted does not work it always takes the background color from [here](https://github.com/odoo/odoo/blob/e20263c4bba3e3cff5b8c7d68f07ce6c7e0bd9aa/addons/web/static/src/views/fields/badge/badge_field.scss#L6) However, since that has now been removed, it no longer applies any background color.in both the case it takes the class text-bg-muted but it is no longer useful.

Reply:
@omra-odoo @mano-odoo We should probably re-implement it on our side. We'll have the same problem with `text-muted` as it has been deprecated (will disappear in bootstrap 6): https://getbootstrap.com/docs/5.3/utilities/colors/

Reply:
@omra-odoo @aab-odoo We should indeed remove the `text-bg-muted` as it's not a valid utility but I don't see why it invalidates your previous solution. Can't we apply another similar background utilities instead of the `text-bg-muted` or remove the utility and let it fallback to its default when `decorationName` is undefined to `text-bg-300` ?

The `text-muted` is still widely used in our codebase and currently refers to `$o-main-color-muted` (see `primary_variables.scss`) instead of the default `--bs-secondary-color` bootstrap suggests. So as it is custom, it will still work until we decide to remove its definition for bootstrap default. Unless I'm misunderstanding something?

Reply:
@mano-odoo `text-bg-muted` classnames are built from what is written in view archs (`decoration-muted`). We don't want to change the decoration API, so we'll still allow to define `decoration-muted` attributes. We can do 2 things to fix it though:
 1) special case `muted` when generating the `text-bg-` classname, to generate a classname that is supported in bs5
 2) define rules in scss for the `text-bg-muted` classname. As we still support `text-muted` (with a bs custo), I would have done the same for `text-bg`. But I don't have a strong opinion on that matter.

Reply:
@aab-odoo Ok understood! Sorry for the confusion and thanks a lot for clarifying. 

Since the current `text-bg-muted` were applied without any styling through the decoration API (because the utility doesn't exist) I'd suggest making a special case to apply the current fallback value which is `text-bg-300` in the `decoration-muted` cases (since it was previously relying on the styling defined on the .badge class in `webclient.scss`). 

Defining a `text-bg-muted` utility could also work but it would feel counter intuitive since it's not in bootstrap `$theme-color` map by default and it's also not a plain color becauses it uses an opacity we will need to adapt it further to ensure contrasting color. Additionally it will change the current visual result of our decoration-muted badges. To avoid this we could define it in css to the same value than `text-bg-300` but it feels more like a workaround since it makes sense only in the cases of the decoration-muted which is referring to an utility that wasn't meant to exist. 

Also I noticed the decoration API defines a text-color on the `o_field_badge`? Is there cases where we display some text in the o_field_badge but outside of the `.badge` ? Since the `text-bg-x` utilities already defines a bg-color and its contrasting color on the badge itself it felt unnecessary.

Reply:
> Thus I'd go for number 1

Alright, fine by me :-)

> Also I noticed the decoration API defines a text-color on the o_field_badge? Is there cases where we display some text in the o_field_badge but outside of the .badge ? Since the text-bg-x utilities already defines a bg-color and its contrasting color on the badge itself it felt unnecessary.

Indeed... I don't think I ever noticed it. The `text-bg` classnames are applied by the implementation of the badge field itself. The `text` classnames are applied by the framework (generic code, no matter which field it is). I guess I didn't know because it doesn't break anything as it's overruled by the `text-bf` classname anyway. So as you said it's just useless. Would be better if we wouldn't apply it obviously, but I don't see a trivial and clean way to do it.

Reply:
@aab-odoo the requested changes have been implemented.Could you please verify?
Thank you for suggestions!